soc/arm/nordic_nrf/nrf52/soc.c

```diff
@@ -47,11 +47,34 @@ extern void z_arm_nmi_init(void);
 #define LOG_LEVEL CONFIG_SOC_LOG_LEVEL
 LOG_MODULE_REGISTER(soc);
 
+#define SOC_MCUBOOT_RESET_TYPE_Pos (0UL)
```

Comment:
Needs documentation to explain what bits are used/reserved (for what features/Kconfigs) and which bits the user can use for their own purposes